--- a/makefile.in	2010-09-23 13:00:42.992049806 +0200
+++ b/makefile.in	2010-09-23 13:03:31.060802259 +0200
@@ -1,8 +1,8 @@
 # @configure_input@
 
 CC = @CC@
-CFLAGS = @CFLAGS@
-CPPFLAGS = @CPPFLAGS@
+CFLAGS = @CFLAGS@ @CPPFLAGS@
+LDFLAGS = @LDFLAGS@
 LIBS = @LIBS@
 
 ifeq (@GCC@,yes)
@@ -17,7 +17,7 @@
 LIBFLAGS = -ltermcap
 
 teco:	$(OBJS)
-	$(CC) -o teco $(OBJS) $(LIBS)
+	$(CC) $(LDFLAGS) -o teco $(OBJS) $(LIBS)
 
 tecbuf.o: tecbuf.c $(HEADERS)
 	$(CC) $(CFLAGS) -c tecbuf.c
